riscv-executor/src/lib.rs

```diff
@@ -699,28 +752,38 @@ mod builder {
             self.set_next_pc().and(Some(st_line))
         }
 
-        pub(crate) fn set_mem(&mut self, addr: u32, val: u32, step: u32, selector: u32) {
+        pub(crate) fn set_mem(&mut self, addr: u32, val: u32, step: u32, selector: u8) {
+            self.submachine_op(
+                MachineInstance::main_memory,
+                Some(selector),
+                &[1.into(), addr.into(), step.into(), val.into()],
+                &[],
+            );
             if let ExecMode::Trace = self.mode {
                 self.trace.mem_ops.push(MemOperation {
                     row: self.trace.len,
                     kind: MemOperationKind::Write,
                     address: addr,
                 });
-                self.memory_machine.write(step, addr, val.into(), selector);
             }
 
             self.mem.insert(addr, val);
         }
 
-        pub(crate) fn get_mem(&mut self, addr: u32, step: u32, selector: u32) -> u32 {
+        pub(crate) fn get_mem(&mut self, addr: u32, step: u32, selector: u8) -> u32 {
             let val = *self.mem.get(&addr).unwrap_or(&0);
+            self.submachine_op(
+                MachineInstance::main_memory,
+                Some(selector),
+                &[0.into(), addr.into(), step.into(), val.into()],
+                &[],
+            );
```

Comment:
same here

Reply:
`submachine_op` already does the test so its not "wrong", but indeed, i'll move it inside the test